src/app/src/sensorUtils.js

```diff
@@ -17,46 +29,87 @@ export function makeRiverGaugeRequest(id, isApiRequest) {
     return axios.get(url);
 }
 
-function parseCsvString(csvString) {
-    const data = Papa.parse(csvString, { header: true, comments: "#", dynamicTyping: true, skipEmptyLines: true });
-    return data;
+export function parseCsvString(csvString) {
+    const data = Papa.parse(csvString, {
+        header: true,
+        comments: '#',
+        dynamicTyping: true,
+        skipEmptyLines: true,
+    });
+
+    return data.data;
 }
 
-export function parseRiverGaugeApiData(id, apiData) {
-    const {
-        data: {
-            value: {
-                timeSeries: data,
-            },
+export function parseRiverGaugeApiData(id, data) {
+    const extractedVariableData = VARIABLES.reduce(
+        (acc, variable, idx) => {
+            const apiVariableData = data[idx];
+            if (apiVariableData) {
+                const sensorValue = Number(
+                    apiVariableData.values[0].value[0].value
+                );
+                return sensorValue !== apiVariableData.variable.noDataValue
+                    ? Object.assign(acc, { [variable]: sensorValue })
+                    : Object.assign(acc, { [variable]: 0 });
```

Comment:
Heads up that this will mutate `acc`. It may be ok to do that here, but you can avoid it altogether -- and any unanticipated problems -- by writing either

```
Object.assign({}, acc, { ...
```

or

```
{ ...acc, [variable]: sensorValue }
```

or using the `update` function from immutability helper.

Reply:
This was a comment on the previous PR that set this up. It doesn't matter IMO if we mutate `acc` since it's just a middle step/convenience.

Reply:
Going to leave as-is since this is only showing up as changed because of Prettier formatting and I didn't modify it otherwise. FWIW, I also think mutating the accumulator is fine.